history.go

```diff
@@ -0,0 +1,75 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrHistoryExisted = errors.New("[merkletree] trying to override existing history")
+	ErrBadEpoch       = errors.New("[merkletree] next epoch's STR has bad epoch ")
+)
+
+type History struct {
+	tree          *MerkleTree
+	currentSTR    *SignedTreeRoot
+	epochInterval int64
+}
```

Comment:
> Is the right name ``PAD`` then? (being a persistent authenticated dictionary, which allows queries against older versions of the structure, to quote that paper). The developer would instantiate a new ``PAD``, which would in turn construct the Merkle tree as part of its internals.

Yeah, I think ``PAD`` is probably the right name for our Merkle tree + STR hash chain construction. But I'm not sure if PAD is the right name for what ``History`` represents right now. Going through the PAD paper, they seem to use the term "snapshot" for each version of the tree (see sections 2 and 3.2). But in essence, the STR already *is* a snapshot of the Merkle tree, plus some additional data, so I actually don't think we need a separate struct for ``History``. We can still form a linked list of STRs if that's the structure we prefer, though the hash chain doesn't technically require a linked list (i.e., we could just store each STR in an array, much like the "snapshot array" described in the PAD paper).